woocommerce/woocommerce
# Added Firefox support for making Order Summary sticky when not longer than view

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Changes shipped on https://github.com/woocommerce/woocommerce/pull/47680 make use of `computedStyleMap` which is [not supported in Firefox](https://caniuse.com/mdn-api_element_computedstylemap)
This PR prevents the console errors and ads support for Firefox by using `getComputedStyles`.


Closes #49525



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. **On Firefox** add a product to the Cart
2. Go to the Checkout Blocks page and open the inspector
3. Verify the following error is not displayed in the JS console `Uncaught TypeError: e.computedStyleMap is not a function...`
4. Reduce the window height so that it's smaller than the order summary.
5. Try scrolling, the order summary should not be sticky.
6. Collapse the items in order summary and expand the window height.
7. Try scrolling now, the order summary should be sticky.

#### Verify everything still works on other supported browsers by following steps:

1. Add some items to cart and go to Checkout.
2. Reduce the window height so that it's smaller than the order summary.
3. Try scrolling, the order summary should not be sticky.
4. Collapse the items in order summary and expand the window height.
5. Try scrolling now, the order summary should be sticky.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
